File: internal/domain/message.go
package domain

// MessageStatus represents the delivery lifecycle of a message.
type MessageStatus string

const (
	StatusSending   MessageStatus = "sending"
	StatusSent      MessageStatus = "sent"
	StatusDelivered MessageStatus = "delivered"
	StatusRead      MessageStatus = "read"
	StatusFailed    MessageStatus = "failed"
)

// Message represents a single chat message.
type Message struct {
	ID        string        `json:"id"`
	ChatID    string        `json:"chatID"`
	SenderID  string        `json:"senderID"`
	Content   string        `json:"content"`
	Timestamp int64         `json:"timestamp"`
	Status    MessageStatus `json:"status"`
}

File: frontend/src/types/message.ts
// Plain data interface matching domain.Message shape.
export interface Message {
  id: string;
  chatID: string;
  senderID: string;
  content: string;
  timestamp: number;
  status: string;
}

export const MessageStatus = {
  Sending: "sending",
  Sent: "sent",
  Delivered: "delivered",
  Read: "read",
  Failed: "failed",
} as const;

export type MessageStatus = (typeof MessageStatus)[keyof typeof MessageStatus];
